### PR TITLE
Deployment, shared deployment & shared class.

### PR DESCRIPTION
**Pull request by Sebastian-Fransson**
- Does not highlight deployment.
- Does not hightlight classes in shared session.

### CODE REVIEW

File: Server/src/diagramcoordinator.erl [hunk 1]
```diff
@@ -12,55 +12,67 @@ init(_Coordinator, _Did, {_, []}, _) -> no_messages;
 %Spawns and Initializes the diagram coordinator
 init(Coordinator, Did, {Classes, Messages}, Class_names) -> 
 	%Sending information that the Coordinator has been spawned. To be printed client-side.
+	io:format("no1"),
 	Coordinator ! {Did, print_information, ["Spawned diagram coordinator"]},
+	io:format("no2"),
 	Pids = spawn_nodes(Classes, Did, Coordinator),
-	loop(Coordinator, Did, Pids, Messages, 1, [], none, Class_names). 
+	io:format("YEEEEES"),
```
Comment (IsabelleTornqvist) [inline Server/src/diagramcoordinator.erl]: Some print statements that should be removed, just a reminder

File: Application/src/model/Parser.java [hunk 2]
```diff
@@ -63,8 +63,10 @@ public void parseDeploymentDiagram(String inputJSON){
 
             //Formating the parsed diagram
             StringBuilder deployString = new StringBuilder();
-
              deployString.append("{");
+             deployString.append("deployment_diagram,");
```
Comment (kosara-g) [inline Application/src/model/Parser.java:67]: version tag?

File: Server/src/lobby.erl [hunk 3]
```diff
@@ -50,7 +50,8 @@ loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID) ->
 	  Format_result = io_lib:format("~p", [{Diagram_ID, Class_names}]) ++ "~",
       send_messages(Members, Format_result),
 	  Self = self(),
-	  loop(Creator_Socket, Password, Members, [{Diagram_ID, Class_names, spawn(fun () -> diagramcoordinator:init(Self, Diagram_ID, {Classes, Messages}) end)}| Diagrams], Lobby_ID);
+	  io:format("Hello im now spawning diagram ~n"),
+	  loop(Creator_Socket, Password, Members, [{Diagram_ID, Class_names, spawn(fun () -> diagramcoordinator:init(Self, Diagram_ID, {Classes, Messages}, Class_names) end)}| Diagrams], Lobby_ID);
```
Comment (kosara-g) [inline Server/src/lobby.erl]: version tag

File: Server/src/lobbycoordinator.erl [hunk 4]
```diff
@@ -55,8 +55,25 @@ loop(Rooms, Lobby_increment) ->
...
+	  loop(Rooms, Lobby_increment);
+	  
+	{Creator_Socket, {class_diagram, Sid, Did, Classes, Relations}} ->
+	  %Get lobby and send deployment through.
+	  case find_room(Rooms, list_to_integer(get_lobby_ID(Sid))) of 
+	    not_created -> no_lobby_created;
+		Pid         -> Pid ! {class_diagram, Creator_Socket, {class_diagram, Sid, Did, Classes, Relations}}
+	  end,
+	  loop(Rooms, Lobby_increment);
+
```
Comment (kosara-g) [inline Server/src/lobbycoordinator.erl:73]: version, again

File: Server/src/usercoordinator.erl [hunk 5]
```diff
@@ -66,24 +66,49 @@ loop(Socket, Diagrams) ->
...
 	  Format_result = io_lib:format("~p", [Diagram]) ++ "~",
 		gen_tcp:send(Socket, Format_result),
-		loop(Socket, Diagrams)
-	  
+		loop(Socket, Diagrams);
+		
+	%Sends a deployment diagram client-side.
+	{deployment_diagram, DeploymentDiagram} ->
+      %Formats the message into a sendable format.
+	  io:format("Who dis"),
```
Comment (IsabelleTornqvist) [inline Server/src/usercoordinator.erl]: More prints - that are helpfull when testing - but probably should be removed before merge.

**Comment by IsabelleTornqvist** [comment]
Tested and received String with diagram content. A bunch of print statements that were helpfull when testing, but should probably not be in the demo tomorrow.

**Review by IsabelleTornqvist — APPROVED** [review]

**Review by kosara-g — APPROVED** [review]

**Pull request merged**